mousestyles/ultradian/__init__.py

```diff
@@ -0,0 +1,381 @@
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: int
+        nonnegative integer indicating the mouse number
+    feature: {"AS", "F", "M_AS", "M_IS", "W", "Distance"}
+        "AS": Active state probalibity
+        "F": Food consumed (g)
+        "M_AS": Movement outside homebase
+        "M_IS": Movement inside homebase
+        "W": Water consumed (g)
+        "Distance": Distance traveled
+    bin_width: int
+        number of minutes, the time interval for data aggregation
+    period_length: int
+        number of hours, usually the significant period
+        length indicated by Lomb-scargle model
+
+    Returns
+    -------
+    res: statsmodel seasonal decomposition object
+        seasonal decomposition result for the mouse.
+        Check the seasonal decomposition plot by res.plot(),
+        seasonl term and trend term by res.seasonal and
+        res.trend separately.
+
+    Examples
+    --------
+    >>> res = seasonal_decomposition(strain=0, mouse=0, feature="W",
+                                     bin_width=30, period_length = 24)
+    """
+    freq = int(period_length * 60 / bin_width)
+    if feature == "Distance":
+        ts = aggregate_movement(strain=strain, mouse=mouse,
+                                bin_width=bin_width)
+    else:
+        ts = aggregate_interval(strain=strain, mouse=mouse,
+                                feature=feature, bin_width=bin_width)
+    res = sm.tsa.seasonal_decompose(ts.values, freq=freq, model="additive")
+    return(res)
+
+
+def strain_seasonal(strain, mouse, feature, bin_width, period_length):
+    """
+    Use seansonal decomposition model on the time series
+    of specified strain, mouse, feature and bin_width.
+    return the seasonal term and the plot of seasonal term
+    by mouse of a set of mouses in a strain
+
+    Parameters
+    ----------
+    strain: int
+        nonnegative integer indicating the strain number
+    mouse: list, set or tuple
+        nonnegative integer indicating the mouse number
+    feature: {"AS", "F", "M_AS", "M_IS", "W", "Distance"}
+        "AS": Active state probalibity
+        "F": Food consumed (g)
+        "M_AS": Movement outside homebase
+        "M_IS": Movement inside homebase
+        "W": Water consumed (g)
+        "Distance": Distance traveled
+    bin_width: int
+        number of minutes, the time interval for data aggregation
+    period_length: int
+        number of hours, usually the significant period
+        length indicated by Lomb-scargle model
+
+    Returns
+    -------
+    seasonal_all: numpy array containing the seasonal term for every
+        mouse indicated by the input parameter
+    seasonal_plot: plot of seasonal term by mouse
+
+    Examples
+    --------
+    >>> res = strain_seasonal(strain=0, mouse={0, 1, 2, 3}, feature="W",
+                              bin_width=30, period_length = 24)
+    """
+    # seasonal decomposition
```

Comment:
Add input check for features (in given set). Add input check for strain mouse (positive integer). Add input check for bin_width(min and max range), etc.